sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -379,22 +383,31 @@ private[hive] class HiveClientImpl(
           s"Hive ${version.fullVersion} does not support altering database location")
       }
     }
-    client.alterDatabase(
+    val props = database.properties
+    val dbOwner = props.get(PROP_OWNER_NAME).filter(_.nonEmpty).getOrElse(userName)
+    val dbOwnerType = props.get(PROP_OWNER_TYPE).filter(_.nonEmpty).getOrElse("USER")
+
+    val hiveDb = new HiveDatabase(
```

Comment:
shall we create a `toHiveDatabase` method for the code here? to remove duplicated code

Reply:
ok